torch/_dynamo/side_effects.py

```diff
@@ -75,6 +75,26 @@ def __init__(self, id_to_variable=None, store_attr_mutations=None, keepalive=Non
         self.store_attr_mutations = store_attr_mutations or collections.OrderedDict()
         self.keepalive = keepalive or []
 
+    def __eq__(self, other: object) -> bool:
+        assert isinstance(other, SideEffects)
+        # NB: do NOT test keepalive
+        return (
+            self.id_to_variable == other.id_to_variable
+            and self.store_attr_mutations == other.store_attr_mutations
+        )
+
+    def diff(self, other: "SideEffects") -> Optional[str]:
```

Comment:
j/w why no prefix on this diff?

Reply:
Is it because it's top-level? is that the distinction? The prefix is for nesting `diff` calls?

Reply:
because I didn't need it; this is called at top level